File: crazychat/lib/crazychat.rb
require "crazychat/ui"
require "crazychat/chat"
require "crazychat/server"

module CrazyChat

  def self.start(address, port, name)
    chat_client = Chat.new("http://#{address}:#{port}", name)

    Server.set :chat_client, chat_client
    Server.set :port, port
    Thread.new { Server.run! }

    CrazyChat.input_loop(chat_client)
  end

end

File: crazychat/CrazyChat.cs
namespace crazychat
{
    using Microsoft.AspNetCore.Hosting;
    using System;
    using System.IO;
    using System.Text;

    public class Program
    {
        public static void Main(string[] args)
        {
            Console.OutputEncoding = Encoding.UTF8;

            var address = "http://localhost";
            var port = args[0];
            var userName = args[1];
            var adr = $"{address}:{port}";
            var host = new WebHostBuilder()
                .UseContentRoot(Directory.GetCurrentDirectory())
                .UseKestrel()
                .UseUrls(adr)
                .UseStartup<Startup>()
                .Build();
            
            using (host) {
                host.Start();
                var server = host.Services.GetService(typeof(ICrazyChatServer)) as ICrazyChatServer;
                new CrazyChatClient(userName, $"{adr}/crazychat", server).InputLoop();
            }
        }
    }
}